src/heaps/fibonacci_heap.jl

```diff
@@ -0,0 +1,367 @@
+        parent.child.right = node
+    end
+end
+
+# extract (delete) the min node from the heap in O(log n) time
+function extractMin(root::MutableFibonacciHeap)
+    minimumNode = root.minNode
+    if minimumNode !== nothing
+        if minimumNode.child !== nothing
+            # attach child nodes to root list
+            allChild = aslist(minimumNode.child)
+            for child in allChild
+                _merge_with_root_list(root, child)
+                child.parent = nothing
+            end
+        end
+        
+        _remove_from_root_list(root, minimumNode)
+        # set new min node in heap
+        if minimumNode == minimumNode.right
+            root.minNode = root.rootList = nothing
+        else
+            root.minNode = minimumNode.right
+            consolidate(root)
+        end
+        root.nodeCount -= 1
+    end
+    return minimumNode.value
+end
+
+# actual linking of one node to another in the root list
+# while also updating the child linked list
+function heaplink(root::MutableFibonacciHeap{T}, y::MutableFibonacciHeapNode{T}, x::MutableFibonacciHeapNode{T}) where T
+    _remove_from_root_list(root, y)
+    y.left = y.right = y
+    _merge_with_child_list(root, x, y)
+    x.degree += 1
+    y.parent = x
+    y.mark = false
+end
+
+# combine root nodes of equal degree to consolidate the heap
+# by creating a list of unordered binomial trees
+function consolidate(root::MutableFibonacciHeap{T}) where T
+    nodes = aslist(root.rootList)
+    len = length(nodes)
+    A = fill!(Vector{Union{MutableFibonacciHeapNode, Nothing}}(undef, Int(round(log(root.nodeCount)) * 2)), nothing)
+    for x in nodes
+        d = x.degree + 1
+        while A[d] !== nothing
+            y = A[d]
+            if x.value > y.value
+                x, y = y, x
+            end
+            heaplink(root, y, x)
+            A[d] = nothing
+            d += 1
+        end
+        A[d] = x
+    end
+    for nod in A
+        if nod !== nothing && nod.value < root.minNode.value
+            root.minNode = nod
+        end
+    end
+end
+
+
+# cut this child node off and bring it up to the root list
+function cut(root::MutableFibonacciHeap{T}, x::MutableFibonacciHeapNode{T}, y::MutableFibonacciHeapNode{T}) where T
+    _remove_from_child_list(root, y, x)
+    y.degree -= 1
+    _merge_with_root_list(root, x)
+    x.parent = nothing
+    x.mark = false
+end
+
+# cascading cut of parent node to obtain good time bounds
+function cascadingcut(root::MutableFibonacciHeap{T}, y::MutableFibonacciHeapNode{T}) where T
+    pNode = y.parent
+    if pNode !== nothing
+        if y.mark == false
+            y.mark = true
+        else
+            cut(root, y, pNode)
+            cascadingcut(root, pNode)
+        end
+    end
+end
+
+# Get list of nodes attached to head (of a circular doubly linked list)
+function aslist(head::MutableFibonacciHeapNode)
+    nodelist, node, stop = MutableFibonacciHeapNode[], head, head
```

Comment:
The type of `MutableFibonacciHeapNode[]` is not fully specified, since the type parameter is unspecified. You can do `typeof(head)[]` instead. This impacts performance.
You also have un-specified types in lines 224 and 282.